doc/source/user/virtual_file_systems.rst

```diff
@@ -625,7 +625,28 @@ The :config:`CPL_VSIS3_CREATE_DIR_OBJECT` configuration option can be set to NO
 
 Starting with GDAL 3.5, profiles that use IAM role assumption (see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-role.html) are handled. The ``role_arn`` and ``source_profile`` keywords are required in such profiles. The optional ``external_id``, ``mfa_serial`` and ``role_session_name`` can be specified. ``credential_source`` is not supported currently.
 
-.. versionadded:: 2.1
+.. _vsis3_imds:
+
+/vsis3/ and AWS Instance Metadata Service (IMDS)
+++++++++++++++++++++++++++++++++++++++++++++++++
+
+On EC2 instances, GDAL will try to use the `IMDSv2 <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html>`__ protocol in priority to get the authentication tokens for AWS S3, and fallback to IMDSv1 in case of failure. Note however that on recent Amazon Linux instances, IMDSv1 is no longer accessible, and thus IMDSv2 must be correctly configured (and even if IMDSv1 is available, mis-configured IMDSv2 will cause delays in the authentication step).
+
+There are known issues when running inside a Docker instance in a EC2 instance that require extra configuration of the instance. For example, you need to `increase the hop limit to 2 <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html#imds-considerations>`__
+
+There are several ways to do this. One way is to run this command:
+::
+
+    aws ec2 modify-instance-metadata-options \
+        --instance-id <instance_id> \
+        --http-put-response-hop-limit 2 \
+        --http-endpoint enabled
+
+Another is to set the HttpPutResponseHopLimit metadata on an AutoScalingGroup LaunchTemplate:
+- https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_InstanceMetadataOptionsRequest.html
+- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-metadataoptions.html
+
+Another possibility is to start the Docker container with host networking (``--network=host``).
```

Comment:
Docker host networking should maybe be frowned upon. It grants your container access to _all_ the ports of the host. 

>  from a security perspective, Docker containers can listen on many ports, even though they only advertise (expose) a single port. Normally this is fine as you only forward the single expected port, however if you use --net=host then you'll get all the container's ports listening on the host, even those that aren't listed in the Dockerfile. This means you will need to check the container closely (especially if it's not yours, e.g. an official one provided by a software project) to make sure you don't inadvertently expose extra services on the machine. [source](https://stackoverflow.com/a/57051970/40785)

If you had two different containers running on one host with `--network=host` I think they could access each others ports.  This could happen on large instances that a scheduler like ECS or Kubernetes can assign many containers to run on a single host.  Because of this I think increasing the HTTP Put Response Hop Limit might be a better solution.

Reply:
> Docker host networking should maybe be frowned upon

thanks. I've added a note about this